pkg/subctl/lighthouse/dns/ensure.go

```diff
@@ -0,0 +1,150 @@
+/*
+© 2020 Red Hat, Inc. and others.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lighthousedns
+
+import (
+	"fmt"
+	"time"
+
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	clientsetifc "k8s.io/client-go/kubernetes/typed/apps/v1"
+	"k8s.io/client-go/rest"
+
+	"github.com/submariner-io/submariner-operator/pkg/internal/cli"
+)
+
+const (
+	operatorImage           = "quay.io/aswinsuryan/lighthouse-cluster-dns-operator:v0.0.1"
+	coreDNSImage            = "quay.io/submariner/lighthouse-coredns:v0.0.1"
+	deploymentCheckInterval = 5 * time.Second
+	deploymentWaitTime      = 2 * time.Minute
+)
+
+func Ensure(status *cli.Status, config *rest.Config) error {
+	clientSet, err := clientset.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	// Disable ClusterVersionOperator
+	_, err = scaleDeployment(clientSet.AppsV1().Deployments("openshift-cluster-version"), "cluster-version-operator", 0)
```

Comment:
Isn't this very openshift specific?, what about others? Can we add comments about the reasoning for this?